en/middleware/uxrce_dds.md

```diff
@@ -438,7 +439,96 @@ Each (`topic`,`type`) pairs defines:
 4. The message type (`VehicleOdometry`, `VehicleStatus`, `OffboardControlMode`, etc.) and the ROS 2 package (`px4_msgs`) that is expected to provide the message definition.
 
 You can arbitrarily change the configuration.
-For example, you could use different default namespaces or use a custom package to store the message definitions. 
+For example, you could use different default namespaces or use a custom package to store the message definitions.
+
+## Fast-RTPS to uXRCE-DDS Migration Guidelines
+
+These guidelines explain how to migrate from using PX4 v1.13 [Fast-RTPS](../middleware/micrortps.md) middleware to PX4 v1.14 `uXRCE-DDS` middleware.
+These are useful if you have [ROS 2 applications written for PX4 v1.13](https://docs.px4.io/v1.13/en/ros/ros2_comm.html), or you have used Fast-RTPS to interface your applications to PX4 [directly](https://docs.px4.io/v1.13/en/middleware/micrortps.html#agent-in-an-offboard-fast-dds-interface-ros-independent).
+
+:::note
+This section contains migration-specific information.
+You should also read the rest of this page to properly understand uXRCE-DDS.
+:::
+
+
+#### Dependencies do not need to be removed
```

Comment:
This was my takeaway. I restructured to make it clear that this is optional. Then I don't need to think about Java.